### PR TITLE
ci(dp): DP Makefile and Actions Workflow  improvements

### PR DESCRIPTION
**Pull request by sklgromek**
## Summary
- Makefile targets cleanup was made
- Makefile target `integration_tests` was created for integration tests as a replacement for triggering them with `CI=true` env variable
- Adding ` _switch_k8s_context` target as protection, to running only on miniukube context

### CODE REVIEW

File: dp/tools/deployment/vendor/elasticsearch.yml [hunk 1]
```diff
@@ -24,6 +24,12 @@ spec:
           env:
             - name: discovery.type
               value: single-node
+          readinessProbe:
```
Comment (jkmar) [inline dp/tools/deployment/vendor/elasticsearch.yml:27]: Let's maybe investigate flake first before proposing this change, I'm not sure if it helps

**Review by jkmar — APPROVED** [review]

File: dp/tools/deployment/vendor/elasticsearch.yml (hunk 1 above)
Comment (xbend) [reply dp/tools/deployment/vendor/elasticsearch.yml:27]: While it was not proven explicitly (we did not notice the integ tests failing) we did see an indication that this fix removes the flakiness. We saw that fluentd-forward did have problems connecting to ES, and entered a long backoff connection retry.
The test managed to succeed, but we did see backoff timeouts kicking in on fluentd-forward due to ES unavailability.

**Review by xbend — APPROVED** [review]

**Comment by xbend** [comment]
@magma/approvers-ci 
This PR is now ready for review and merge. It should solve most if not all problems with flaky Domain Proxy integration test runs so the quicker we get it into master, the better.

File: .github/workflows/dp-workflow.yml [hunk 2]
```diff
@@ -390,6 +409,8 @@ jobs:
           path: |
             ${{ env.TEST_DIR }}
             /tmp/${{ matrix.test_type }}.txt
+            /tmp/k8s-pods-logs
```
Comment (nstng) [inline .github/workflows/dp-workflow.yml:412]: NIT: is this more robust if results files are collected in a dedicated /tmp/foo folder and this folder is uploaded? I.e., omit to list all files created in different places.

**Review by nstng — APPROVED** [review]
CI changes (dp-workflow.yml) lgtm - artifacts are uploaded successfully.

File: .github/workflows/dp-workflow.yml (hunk 2 above)
Comment (sklgromek) [reply .github/workflows/dp-workflow.yml:412]: Good point. I will improve it in another PR.

**Pull request merged**